shared/src/simulation/behaviors/assign-leader.ts

```diff
@@ -110,6 +110,16 @@ export const assignLeaderBehavior: SimulationBehavior<AssignLeaderBehaviorState>
                         }
                     }
                     break;
+                case 'personnelRemovedEvent':
+                    {
+                        // if the leader is removed from the region the value is unset
+                        // a new leader will be chosen in the next tick
+
+                        if (event.personnelId === behaviorState.leaderId) {
+                            behaviorState.leaderId = undefined;
+                        }
+                    }
+                    break;
```

Comment:
Please use the `changeLeader` helper and add a leader in the same tick as some activities behave specifically to the absence of a leader.

Reply:
I extracted the new leader selection to a helper